sktime/tests/test_all_estimators.py

```diff
@@ -879,7 +879,7 @@ def test_constructor(self, estimator_class):
 
         # Ensure that each parameter is set in init
         init_params = _get_args(type(estimator).__init__)
-        invalid_attr = set(init_params) - set(vars(estimator)) - {"self"}
+        invalid_attr = set(init_params) - set(dir(estimator)) - {"self"}
```

Comment:
with `attrs`, `vars` seems not to work properly.  I think this has something to do with it not playing well with the `__dict__` function that `vars` seems to rely on.  They do get added as valid attributes in the `__init__` however, which is captured by using `dir` instead, and I think is what we are really interested in???

They are slightly different functions though, as is summarized a bit [here](https://stackoverflow.com/questions/980249/difference-between-dir-and-vars-keys-in-python).  Let me know if people think this is a reasonable change or not.  (I believe it should still execute this test as expected for other estimators, ie if something is passed in init, but is not added as an attribute then this test**shoudl**still fail as expected).

Reply:
could we separate out the `attrs` business from this PR? In a separate PR?

I think that's a different issue - modification of the base object interface.

The separate PR should be accompanied by tests and dummy examples for the `attr` variant.

Reply:
Yea ok, I will leave this PR as the `attrs` PR but will open another HMM wrapper PR without it.